### PR TITLE
SAK-43110 show rubric button when in lessons -> samigo

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-43110

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/BeginDeliveryActionListener.java [hunk 1]
```diff
@@ -101,11 +101,12 @@ public void processAction(ActionEvent ae) throws
       // e.g. take assessment via url, actionString is set by LoginServlet.
       // preview and take assessment is set by the parameter in the jsp pages
       delivery.setActionString(actionString);
-      if ("reviewAssessment".equals(actionString) || "takeAssessment".equals(actionString)) {
-        delivery.setRbcsToken(rubricsService.generateJsonWebToken(RubricsConstants.RBCS_TOOL_SAMIGO));
-      }
     }
-    
+
+    if ("reviewAssessment".equals(delivery.getActionString()) || "takeAssessment".equals(delivery.getActionString()) || "takeAssessmentViaUrl".equals(delivery.getActionString())) {
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/BeginDeliveryActionListener.java]: StringUtils.equalsAny, I know this pattern is already used in the file but equalsAny is more desirable.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java [hunk 2]
```diff
@@ -1685,11 +1685,16 @@ public ItemGradingData getItemGradingData(String publishedItemId) {
   }
 
   public String getSiteId() {
-    siteId = null;
-    Placement currentPlacement = ToolManager.getCurrentPlacement();
-    if(currentPlacement != null)
-      siteId = currentPlacement.getContext();
-    return siteId;
+
+    if (siteId != null) {
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java]: StringUtils.isNotBlank

File: rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java [hunk 3]
```diff
@@ -192,11 +192,15 @@ private String getCurrentSiteId(String method){
     }
 
     public String generateJsonWebToken(String tool) {
-        return generateJsonWebToken(tool, getCurrentSiteId("generateJsonWebToken"));
+        return generateJsonWebToken(tool, null);
     }
 
     public String generateJsonWebToken(String tool, String siteId) {
 
+        if (siteId == null) {
```
Comment (mpellicer) [inline rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java]: StringUtils.isBlank

File: rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java [hunk 4]
```diff
@@ -192,11 +192,15 @@ private String getCurrentSiteId(String method){
     }
 
     public String generateJsonWebToken(String tool) {
-        return generateJsonWebToken(tool, getCurrentSiteId("generateJsonWebToken"));
+        return generateJsonWebToken(tool, null);
     }
 
     public String generateJsonWebToken(String tool, String siteId) {
 
+        if (StringUtils.isBlank(siteId)) {
```
Comment (ern) [inline rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java]: duplicate logic a few lines down

**Pull request merged**